### PR TITLE
feat: Added single product page and coupon

### PR DESCRIPTION
**Pull request by BalajiHarinath**
Added single product page and coupon. Please review.

### CODE REVIEW

File: src/Pages/Cart/CartDetails.jsx [hunk 1]
```diff
@@ -1,27 +1,90 @@
...
+                        </div>                     
+                    </div>
+
+                    <div className="modal-btn-container">
+                        <button className={`${isOfferEncashed ? "btn-solid btn-small modal-btn poniter-events-none" : "btn-solid btn-small modal-btn"}`} onClick={couponApplyHandler}>{`${isEligibleForCoupon ?  isOfferEncashed ? "Offer Encashed" : "Apply" : "Not Eligible"}`}</button>
+                    </div>
+                    <button className="modal-close-btn" onClick={() => setDisplayCoupon({...displayCoupon, status:!displayCoupon.status})}><i className="fa fa-times" aria-hidden="true"></i></button>
+                </div>
+            </div>
+
```
Comment (deekshithmd) [inline src/Pages/Cart/CartDetails.jsx:86]: Check the UI of price detail card...It is floating over footer

File: src/Pages/SingleProduct/SingleProduct.css [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    width: 20rem;
+    height: inherit;
+    object-fit: contain;
+  }
+  
+  .product-details {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    gap: 0.5rem;
```
Comment (deekshithmd) [inline src/Pages/SingleProduct/SingleProduct.css:26]: you can give this in px so that value remains fixed while changing screen size

**Review by deekshithmd — APPROVED** [review]

File: src/Pages/Cart/Cart.css [hunk 3]
```diff
@@ -62,6 +63,47 @@
     position: relative;
 }
 
+/*Coupon*/
+.modal-container-offer {
+    position: fixed;
+    top: 30vh;
+    left: 30vw;
+    /* filter: drop-shadow(8px 8px 10px rgb(97, 95, 95)) */
```
Comment (sainath7878) [inline src/Pages/Cart/Cart.css]: remove commented code

File: src/Pages/Cart/CartDetails.jsx [hunk 4]
```diff
@@ -1,27 +1,90 @@
...
+    const [totalAmount, setTotalAmount] = useState(0);
+    const [isEligibleForCoupon, setIsEligibleForCoupon] = useState(true);
+    const [isOfferEncashed, setIsOfferEncashed] = useState(false)
+
+    useEffect(() => {     
+        setTotalAmount(getTotalAmount(cart, deliveryFee));      
+    },[removeFromCart, updateItemQuantity])
+
+    useEffect(() => {
+        console.log(totalAmount)
```
Comment (sainath7878) [inline src/Pages/Cart/CartDetails.jsx]: remove console.log

File: src/Pages/Cart/CartDetails.jsx (hunk 1 above)
Comment (BalajiHarinath) [reply src/Pages/Cart/CartDetails.jsx:86]: okay will check this.

**Review by sainath7878 — APPROVED** [review]
If a coupon is already applied, there is no way to change the coupon. Check the functionality. Apart from that, everything else looks good. UI is awesome too!

**Pull request merged**